network-api/networkapi/wagtailpages/pagemodels/buyersguide/products.py

```diff
@@ -960,6 +968,12 @@ def get_primary_related_articles(self) -> models.QuerySet['BuyersGuideArticlePag
     def get_secondary_related_articles(self) -> models.QuerySet['BuyersGuideArticlePage']:
         return self.get_related_articles()[3:]
 
+    def check_for_categories_featuring_cta(self):
```

Comment:
**Question** Have you considered adding tests for this custom method?

Reply:
Hi @tbrlpld, this function is now called "[return_featured_cta](https://github.com/mozilla/foundation.mozilla.org/blob/47d310b7f20ac99fa66f0c0b8a60279ae7d6ac8a/network-api/networkapi/wagtailpages/pagemodels/buyersguide/products.py#L971)".

While trying to add tests that call this method directly, the model_cluster package would return an error regarding the .filter() query that is being called here:

```
raise NotImplementedError("Filter expression not supported: %s" % '__'.join(key_clauses))
NotImplementedError: Filter expression not supported: category__show_cta
```

I tried to troubleshoot for some time with little success.


However, this method's functionality is technically being tested/used in the following other tests:

- [test_product_with_enabled_category_shows_cta](https://github.com/mozilla/foundation.mozilla.org/blob/6b83d7e2a420d201fc60a643af156560cedba5b6/network-api/networkapi/wagtailpages/tests/buyersguide/test_products.py#L237)
- [test_product_with_disabled_category_hides_cta](https://github.com/mozilla/foundation.mozilla.org/blob/6b83d7e2a420d201fc60a643af156560cedba5b6/network-api/networkapi/wagtailpages/tests/buyersguide/test_products.py#L262)
- [test_product_page_cta_with_multiple_categories](https://github.com/mozilla/foundation.mozilla.org/blob/6b83d7e2a420d201fc60a643af156560cedba5b6/network-api/networkapi/wagtailpages/tests/buyersguide/test_products.py#L284)

As these tests take a product page with multiple categories, and check whether or not they return the featured CTA as expected, which would be done with `return_featured_cta`.

I was wondering if this would suffice, or if not, if we can work together to troubleshoot?

Reply:
I think it would be better to test the method directly to avoid side effects. I have responded on Slack.

Reply:
What do you think about calling the method `get_featured_cta` as all other "fetch" methods use `get_` as the start.

Reply:
Hi @tbrlpld, updating the name sounds good to me! I have implemented that.

Regarding your suggestion from slack, I have tried it and unfortunately was greeted with the same error.

After trying a couple of different things, I found the most straightforward (to read) solution I could come up with (though might not be the most optimized) is this 1 liner here:

```
        if ProductPageCategory.objects.filter(product=self, category__show_cta=True):
            return get_featured_cta(self)
        else:
            return None
```

Was wondering what you thought about this? 

Thanks!

Reply:
Hi @tbrlpld, updating the name sounds good to me! I have implemented that.

Regarding your suggestion from slack, I have tried it and unfortunately was greeted with the same error.

After trying a couple of different things, I found the most straightforward (to read) solution I could come up with (though might not be the most optimized) is this 1 liner here:

```
        if ProductPageCategory.objects.filter(product=self, category__show_cta=True):
            return get_featured_cta(self)
        else:
            return None
```

The functionality is the same and the test suite does not seem to mind this filter query.
Was wondering what you thought about this? 

Thanks!

Reply:
@danielfmiranda Yup, I think this is great. Only suggestion would be to still add the `.exists()` to avoid pulling all the data from the database when all we want it a `True`.